src/pages/main.tsx

```diff
@@ -1,59 +1,80 @@
-import React, { useEffect, useRef, useState } from 'react';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
 import Header, { links } from '../components/header';
 import SearchBar from '../components/searchBar';
 import ItemsCardsList from '../components/itemsCardsList';
-import { getItems, IItemData } from '../api/api';
-import Modal, { OVERLAY_ID, CLOSE_ID } from 'components/modal';
-import ItemCard from 'components/itemCard';
+import { getItems, IGetItemsOptions } from '../api/api';
 import DownloadIndicator from 'components/downloadIndicator';
+import GlobalStateContext from 'state/context';
+import { ACTION } from 'state/reducer';
+import SortingSwitcher from 'components/sortingSwitcher';
+import PaginationSwitcher from 'components/paginationSwitcher';
+import { useNavigate } from 'react-router-dom';
 
 function Main() {
-  const [itemsData, setItemsData] = useState([] as IItemData[]);
-  const [isModalVisible, setIsModalVisible] = useState(false);
-  const itemModalIdx = useRef(0);
-  const errMsg = useRef('');
-  const [isDataLoading, setIsDataLoading] = useState(true);
+  const [isDataLoading, setIsDataLoading] = useState(false);
+  const { globalState, dispatch } = useContext(GlobalStateContext);
+  const navigate = useNavigate();
 
-  const getItemsData = async (search: string) => {
-    setItemsData([]);
-    setIsDataLoading(true);
-    const data = await getItems(search);
-    errMsg.current = data.errMsg;
-    setIsDataLoading(false);
-    setItemsData(data.items);
-  };
-
-  const showModal: React.MouseEventHandler = (e) => {
-    itemModalIdx.current = parseInt(e.currentTarget.id);
-    setIsModalVisible(true);
-  };
-
-  const closeModal: React.MouseEventHandler = (e) => {
-    const target = e.target as HTMLElement;
-
-    if (!target || (target && target.id !== OVERLAY_ID && target.id !== CLOSE_ID)) {
-      return;
-    }
+  const getItemsData = useCallback(
+    async (opt: IGetItemsOptions) => {
+      setIsDataLoading(true);
+      const data = await getItems(opt);
+      setIsDataLoading(false);
+      dispatch({
```

Comment:
Please try to be more declarative. You can import some sort of callback from your context (using custom hooks):
`saveItemsData(data)` and encapsulate all mapping there. Hooks were invented for this 👍